tiledb/ml/readers/_tensor_schema.py

```diff
@@ -221,11 +235,15 @@ def max_partition_weight(self) -> int:
         rows_per_slice = dim_tiles[0]
 
         # Reading a slice of `rows_per_slice` rows requires reading a number of tiles that
-        # depends on the size and tile extent of each dimension after the first one.
+        # depends on the size, secondary slice, and tile extent of each dimension after the first one.
         assert len(self.shape) == len(dim_tiles)
-        tiles_per_slice = np.prod(
-            [ceil(size / tile) for size, tile in zip(self.shape[1:], dim_tiles[1:])]
-        )
+        
+        tiles_per_slice = 1
+        for i, tile in enumerate(dim_tiles[1:], 1):
+            if i in self._secondary_slices and isinstance(self._secondary_slices[i], list):
+                tiles_per_slice *= len(set([(x - self._ned[i][0]) // tile for x in self._secondary_slices[i]]))
+            else:
+                tiles_per_slice *=  ceil(self.shape[i] / tile)
```

Comment:
This `else` covers both the original case (without a secondary selector) and the case of non-sequence (slice or single integer) secondary selector. Is this deliberate or forgotten?

Reply:
This is deliberate. All of the non-sequence secondary selectors are contiguous sequences, so the same logic applies (provided that the shape is updated accordingly, which it is in this case).

Reply:
I see. There's still a bug in case the slice start and/or stop don't align with the tile boundaries. You can verify this by running `test_max_partition_weight_dense` with ` {"d1": slice(1,2)}` in `secondary_slices`.